src/readbin/compare.ml

```diff
@@ -0,0 +1,119 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+exception No_unstripped_file
+
+let print_source = function
+  | `bw -> print_endline "bw"
+  | _ -> print_endline "others"
+
+let source = [
+  "BW", `bw;
+  "SymTbl", `symtbl;
+  "User", `user;
+  "Ida", `ida;
+]
+
+let tool : _ Term.t =
+  let doc = "The tool that we are going to evaluate. It can be BW or Ida." in
+  Arg.(value & pos 1 (enum source) `bw & info [] ~docv:"tool" ~doc)
+
+let gt : _ Term.t =
+  let doc = "The ground truth source. One can direct to a user file or use the symbol
+  table from unstripped binary (-u required)." in
+  Arg.(value & pos 2 (enum source) `symtbl & info [] ~docv:"gt" ~doc)
+
+let symsfile : string option Term.t =
+  let doc = "The symbol table of binaries. This requires the binaries to use this file as symbols source." in
+  Arg.(value & opt (some non_dir_file) None & info ["syms"; "s"]
+         ~docv:"syms" ~doc)
+
+let unstrip_bin : string option Term.t =
+  let doc = "The unstripped binary." in
+  Arg.(value & opt (some non_dir_file) None & info ["unstrip"; "u"]
+         ~docv:"unstripped_bin" ~doc)
+
+let print_metrics : _ list Term.t =
+  let opts = [
+    "precision", `with_prec;
+    "recall", `with_recl;
+    "F_measure", `with_F;
+    "TP", `with_TP;
+    "FP", `with_FP;
+    "FN", `with_FN;
+  ] in
+  let doc = "Print metrics. User can choose to print -cprecision, -crecall,
+  -cF_measure, -cTP, -cFP and -cFN." in
+  Arg.(value & opt_all (enum opts) (List.map ~f:snd opts) &
+       info ["with-metrics"; "c"] ~doc)
+
+let bin : string Term.t =
+  let doc = "The testing stripped binary." in
+  Arg.(required & pos 0 (some non_dir_file) None & info [] ~docv:"binary" ~doc)
+
+let func_start bin symsfile unstrip_bin : _ -> Addr.Set.t * string = function
+  | `bw -> Func_start.byteweight bin, "BW"
+  | `user -> (match symsfile with
+      | None -> raise Not_found
+      | Some f -> Func_start.user f, "User")
+  | `symtbl -> (match unstrip_bin with
+      | None -> raise No_unstripped_file
+      | Some b -> Func_start.symbols b, "Symbol")
+  | `ida -> Func_start.ida bin, "IDA"
+  | _ -> Addr.Set.empty, "None"
```

Comment:
this pattern should be refutable, don't add this wildcard, there is no other variants